javascript/javascript2/week1/script.js

```diff
@@ -0,0 +1,90 @@
+the return is, therefore {total: 1, æ: 0, ø: 0, å: 1} */
+const danishString2 = "Blå grød med røde bær";
+countLetters(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1} . yay
+
+// Spirit name Generation
+
+const body = document.querySelector("body");
+const input = document.getElementById("iName"); // grab an input if it has className set to InputName
+const output = document.getElementsByClassName("output"); // grab an output that has this class
+// we assume that elements with such classes will have values that can be considered for our purposes
+const myButton = document.querySelector("button"); // grab the first ( and only, in our case ) button
+const spiritArr = [
+  "Care bear",
+  "Mist Dragon",
+  "Star Cat",
+  "Destiny Rat",
+  "Miracle Mallard",
+  "Gallactic Stag",
+  "Morning-dew Ladybug",
+  "Rainbow Snow Fox",
+  "Questgiving Cat",
+  "Freedom Eagle",
+]; // array of "spirit" animals
+
+/*  I am a bit confused with the part that asks to create functionality where user might request a new spirit animal.
+ The solution below does that by just pressing the generate button again, and honestly, I am unsure how to make it to just generate one, without
+ specifically applying a restriction on doing that (clicking button again) with more code. 
+ Smth like >> if (output[0].textContent !== '') {myButton.disabled = true},  */
+let selector = document.getElementById("genSelector");
+
+function generateSpirit() {
+  console.log(input.value);
+  // onclick event
+  if (input.value == "") {
+    // that checks for empty input
+    alert("please check the input field"); // and alerts user if input is indeed empty
+  } else {
+    // otherwise
+    let i = Math.round(Math.random() * 9); // random number from 0 to 9 ( just what we need for a 10 items long array of spirit-animals)
+    output[0].textContent = input.value + " - " + spiritArr[i]; // make it that text of the output element becomes our necessary output
+  }
+}
+body.onload = function (event) {
+  myButton.addEventListener("click", generateSpirit); // default
+};
+selector.onchange = function (event) {
+  switch (selector.value) {
+    case "click": // there should be a better way ........
+      myButton.addEventListener("click", generateSpirit);
+      input.removeEventListener("mouseenter", generateSpirit);
+      input.removeEventListener("input", generateSpirit);
+      break;
+    case "hover":
+      myButton.removeEventListener("click", generateSpirit);
+      input.addEventListener("mouseenter", generateSpirit);
+      input.removeEventListener("input", generateSpirit);
+      // This will need to tab into to avoid alert, but let's leave it like this :)
+      break;
+    case "change":
+      myButton.removeEventListener("click", generateSpirit);
+      input.removeEventListener("mouseenter", generateSpirit);
+      input.addEventListener("input", generateSpirit);
+      break;
+  }
```

Comment:
Fantastic solution, very well done. If you wanted to make it more concise, you could refactor the "remove event listener" functionality into its own function and call that as the first thing, e.g. something like this: 

```javascript
function clearEventListeners() {
    myButton.onclick = null;
    input.oninput = null;
    input.onmouseover = null;
}
```

Then your onchange function could be something like
```javascript
function updateSelect(event) {
    const selectValue = event.target.value;
    
    clearEventListeners();

    switch(selectValue) {
        case "hover":
            // User wants a spirit animal when hovering the input
            input.onmouseover = setSpiritAnimal
            return;
        case "click":
            // User wants a spirit animal when clicking the button
            button.onclick = setSpiritAnimal;
            return;
        case "written":
            // User wants it to update as it is being written
            input.oninput = setSpiritAnimal
            return;
    }
}
``` 

But yours was a perfectly great solution too.